src/core/server/elasticsearch/client/mocks.ts

```diff
@@ -31,42 +31,56 @@ const createInternalClientMock = (): DeeplyMockedKeys<Client> => {
     '_events',
     '_eventsCount',
     '_maxListeners',
+    'constructor',
     'name',
     'serializer',
     'connectionPool',
     'transport',
     'helpers',
   ];
 
+  const getAllPropertyDescriptors = (obj: Record<string, any>) => {
+    const descriptors = Object.entries(Object.getOwnPropertyDescriptors(obj));
+    let prototype = Object.getPrototypeOf(obj);
+    while (prototype != null && prototype !== Object.prototype) {
+      descriptors.push(...Object.entries(Object.getOwnPropertyDescriptors(prototype)));
+      prototype = Object.getPrototypeOf(prototype);
+    }
+    return descriptors;
+  };
+
   const mockify = (obj: Record<string, any>, omitted: string[] = []) => {
-    Object.keys(obj)
-      .filter((key) => !omitted.includes(key))
-      .forEach((key) => {
-        const propType = typeof obj[key];
-        if (propType === 'function') {
+    const descriptors = getAllPropertyDescriptors(obj);
```

Comment:
When updating `@elastic/elasticsearch` to include the improvements which will be made as part of https://github.com/elastic/elasticsearch-js/pull/1314, I had to significantly change the way that we mock the Client because it's now using prototypical inheritance and getters. @pgayvallet and @restrry, have we previously evaluated switching to using https://github.com/elastic/elasticsearch-js-mock instead?

Reply:
Yea, we have, and `elasticsearch-js-mock` is more of a integration test mock, allowing to mock specific endpoints with 'real' responses. It doesn't 'play well' with jest, as it wouldn't allow to use syntax such as `expect(client.search).toHaveBeenCalled` and would have been a divergence from all other core's mocks.

Reply:
Cool, thanks for the background @pgayvallet :)

Reply:
a comment would be useful here